### PR TITLE
Cknieling/avoid attribute nameclashes

### PR DESCRIPTION
**Pull request by chr11115**
- Add index suffix to internal maya names to avoid possible name clashes due to edge cases in import/style names

### CODE REVIEW

File: src/serlio/modifiers/RuleAttributes.cpp [hunk 1]
```diff
@@ -115,11 +124,15 @@ RuleAttributeSet getRuleAttributes(const std::wstring& ruleFile, const prt::Rule
...
-		p.mayaBriefName = getBriefName(p.fqName);
-		p.mayaFullName = getFullName(p.fqName);
 		p.mayaNiceName = getNiceName(p.fqName);
 		p.mType = attr->getReturnType();
 
+		//make sure maya names are unique
+		const std::wstring mayaBriefName = getBriefName(p.fqName);
+		const std::wstring mayaFullName = getFullName(p.fqName);
+		p.mayaBriefName = mayaBriefName + getDuplicateCountSuffix(mayaBriefName, mayaNameDuplicateCountMap);
+		p.mayaFullName = mayaFullName + getDuplicateCountSuffix(mayaFullName, mayaNameDuplicateCountMap);
```
Comment (mistafunk) [inline src/serlio/modifiers/RuleAttributes.cpp:134]: consider hiding the clash resolution in the brief/full name functions

File: src/serlio/modifiers/RuleAttributes.cpp [hunk 2]
```diff
@@ -56,6 +56,13 @@ std::wstring getNiceName(const std::wstring& fqAttrName) {
 	return cleanForMaya(prtu::removeImport(prtu::removeStyle(fqAttrName)));
 }
 
+std::wstring getDuplicateCountSuffix(const std::wstring& mayaName, std::map<std::wstring, int>& mayaNameDuplicateCountMap) {
```
Comment (mistafunk) [inline src/serlio/modifiers/RuleAttributes.cpp:59]: Please move this function into the utils and write unit tests for it.

**Review by mistafunk — CHANGES_REQUESTED** [review]
LGTM, but I have to insist on some unit tests ;-)

File: src/serlio/modifiers/RuleAttributes.cpp (hunk 1 above)
Comment (chr11115) [reply src/serlio/modifiers/RuleAttributes.cpp]: fixed

File: src/serlio/modifiers/RuleAttributes.cpp (hunk 2 above)
Comment (chr11115) [reply src/serlio/modifiers/RuleAttributes.cpp]: Added unit test based on example in internal ticket

**Review by mistafunk — APPROVED** [review]
LGTM

**Pull request merged**